modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/GridDhtPartitionsStateValidator.java

```diff
@@ -96,13 +92,25 @@ public void validatePartitionCountersAndSizes(
                 ignoringNodes.add(id);
         }
 
+        StringBuilder error = new StringBuilder();
+
+        if (!resultUpdCnt.isEmpty())
+            error.append("Partitions update counters are inconsistent for " + fold(topVer, resultUpdCnt));
+
         if (!cctx.cache().cacheGroup(top.groupId()).mvccEnabled()) { // TODO: Remove "if" clause in IGNITE-9451.
             // Validate cache sizes.
-            result = validatePartitionsSizes(top, messages, ignoringNodes);
+            Map<Integer, Map<UUID, Long>> resultSize = validatePartitionsSizes(top, messages, ignoringNodes);
 
-            if (!result.isEmpty())
-                throw new IgniteCheckedException("Partitions cache sizes are inconsistent for " + fold(topVer, result));
+            if (!resultSize.isEmpty()) {
+                if (error.length() > 0)
+                    error.append(System.lineSeparator());
```

Comment:
I really don't like idea of combing messages in such way. Please consider using message like "Both partitions cache size and update counters..." if both partitions cache size and update counters have diverged. That will probably require rework of `org.apache.ignite.internal.processors.cache.distributed.dht.topology.GridDhtPartitionsStateValidator#fold`